src/Microsoft.ML.StandardLearners/Standard/Online/AveragedPerceptron.cs

```diff
@@ -60,8 +60,13 @@ public sealed class Options : AveragedLinearOptions
             /// <summary>
             /// A custom <a href="tmpurl_loss">loss</a>.
             /// </summary>
-            [Argument(ArgumentType.Multiple, HelpText = "Loss Function", ShortName = "loss", SortOrder = 50)]
-            public ISupportClassificationLossFactory LossFunction = new HingeLoss.Options();
+            [Argument(ArgumentType.Multiple, Name = "LossFunction", HelpText = "Loss Function", ShortName = "loss", SortOrder = 50)]
+            internal ISupportClassificationLossFactory LossFunctionFactory1 = new HingeLoss.Options();
+
+            /// <summary>
+            /// A custom <a href="tmpurl_loss">loss</a>.
```

Comment:
>tmpurl_loss [](start = 34, length = 11)

Were you intending to update this?

Reply:
Also I noted elsewhere you noted what the replacement value would be, which struck me as positive. Will we do that here? (That said, that is merely documentation, which while important, I would not prioritize since the important thing is to get the *shape* of the API right, which you seem to have done here.)

---
In reply to: [263108655](https://github.com/dotnet/machinelearning/pull/2856#discussion_r263108655) [](ancestors = 263108655)

Reply:
Shahab is working on this, issue #2356

---
In reply to: [263109039](https://github.com/dotnet/machinelearning/pull/2856#discussion_r263109039) [](ancestors = 263109039,263108655)